walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -24,62 +24,56 @@
 
 public final class FindMeetingQuery {
     public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-        Set<TimeRange> requiredTimes = new HashSet<>();
-        Set<TimeRange> optionalTimes = new HashSet<>();
+        List<TimeRange> requiredTimes = new ArrayList<>();
+        List<TimeRange> optionalTimes = new ArrayList<>();
         if (request.getDuration() <= TimeRange.END_OF_DAY - TimeRange.START_OF_DAY) {
             // before checking the event availability, the entire day is available
             // return timeslots required attendees are available for
             requiredTimes = determineAvailableTimes(events, request, request.getAttendees());
             // return timeslots optional attendees are available for
             optionalTimes = determineAvailableTimes(events, request, request.getOptionalAttendees());
         }
-        // sort times that required attendees are available for
-        List<TimeRange> finalRequiredTimes = new ArrayList<>();
-        finalRequiredTimes.addAll(requiredTimes);
-        Collections.sort(finalRequiredTimes, TimeRange.ORDER_BY_START);
-        // sort times that optional attendees are available for
-        List<TimeRange> finalOptionalTimes = new ArrayList<>();
-        finalOptionalTimes.addAll(optionalTimes);
-        Collections.sort(finalOptionalTimes, TimeRange.ORDER_BY_START);
-
+        
         // check to see if the lists need to be compared; if there are no attendees at all, the entire 
         // day is available. Otherwise, if there are no optional attendees, we can
         // return times that work for required attendees, and vice versa.
         if (request.getAttendees().size() == 0 && request.getOptionalAttendees().size() == 0) {
```

Comment:
the empty case should come first so you don't have to call determineAvailableTimes

Reply:
also you can use .isEmpty